// @wix/cli/build/chunk-Q2SUFPGY.js
import { createRequire as _createRequire } from 'node:module';
const require = _createRequire(import.meta.url);
import {
  ProjectModelProvider,
  loadProjectModel
} from "./chunk-4ASXFAN7.js";
import {
  HttpClientProvider,
  buildHttpClientHeaders
} from "./chunk-KBORAL57.js";
import {
  PanoramaProvider
} from "./chunk-BVM3FYNM.js";
import {
  addUserFields,
  reportCommandStartEvent
} from "./chunk-ZBDV4YE5.js";
import {
  AccountAuthProvider,
  BiProvider,
  DebugLogProvider,
  ErrorReporterProvider
} from "./chunk-6PYC5DBT.js";
import {
  render
} from "./chunk-LQXRNBJ2.js";
import {
  require_react
} from "./chunk-NRAQAV6T.js";
import {
  getDebugLogFilePath
} from "./chunk-P23NP3OP.js";
import {
  __toESM,
  init_esm_shims
} from "./chunk-4EFJZ3GQ.js";

// ../cli-astro-commands/src/commands/render-command.tsx
init_esm_shims();
var import_react = __toESM(require_react(), 1);
import { cwd } from "node:process";
async function initServices(model, { biLogger, errorReporter }) {
  errorReporter.setContext("Application Info", {
    "Application ID": model.config.appId,
    "Site ID": model.config.siteId
  });
  biLogger.updateDefaults({
    _appId: model.config.appId
  });
}
async function renderCommand(command, { biLogger, errorReporter, panorama }, cliVersion, tree) {
  await runCommand(
    command,
    { biLogger, errorReporter, panorama },
    cliVersion,
    async (model) => {
      biLogger.updateDefaults({ command: command.name() });
      const debugLogFilePath = getDebugLogFilePath(model.projectFolder);
      const handleLogin = (userInfo) => {
        errorReporter.setUser({ id: userInfo.userId });
        addUserFields(biLogger, userInfo.userId);
      };
      const headers = buildHttpClientHeaders({
        flow: "Astro",
        command: command.name(),
        cliVersion
      });
      await render(
        /* @__PURE__ */ import_react.default.createElement(BiProvider, { value: biLogger }, /* @__PURE__ */ import_react.default.createElement(ErrorReporterProvider, { value: errorReporter }, /* @__PURE__ */ import_react.default.createElement(PanoramaProvider, { value: panorama }, /* @__PURE__ */ import_react.default.createElement(AccountAuthProvider, { onLogin: handleLogin }, /* @__PURE__ */ import_react.default.createElement(HttpClientProvider, { headers }, /* @__PURE__ */ import_react.default.createElement(DebugLogProvider, { filePath: debugLogFilePath }, /* @__PURE__ */ import_react.default.createElement(ProjectModelProvider, { value: model }, tree)))))))
      );
    }
  );
}
async function runCommand(command, services, cliVersion, commandFn) {
  const model = await loadProjectModel(cwd());
  await initServices(model, services);
  services.biLogger.updateDefaults({
    command: command.name()
  });
  const commandStatus = reportCommandStartEvent({
    biLogger: services.biLogger,
    command,
    cliVersion,
    flow: "Astro",
    extensions: {},
    extraFields: void 0
  });
  try {
    await commandFn(model);
    commandStatus.succeed();
  } catch (e) {
    commandStatus.failed(e);
    throw e;
  }
}

export {
  renderCommand,
  runCommand
};
