// @wix/cli/build/EnvRemoveCommand-JEE4FQGN.js
import { createRequire as _createRequire } from 'node:module';
const require = _createRequire(import.meta.url);
import {
  useBackendAsAServiceClient
} from "./chunk-4F6VNWLG.js";
import "./chunk-IZPIRKJT.js";
import {
  useProjectModel
} from "./chunk-GJ3FMAEJ.js";
import "./chunk-C6TNYPLG.js";
import "./chunk-BKY3IKB3.js";
import {
  SiteAuthProvider
} from "./chunk-SZ4GCQYL.js";
import "./chunk-WMCHVSMU.js";
import "./chunk-6C4RLYLU.js";
import "./chunk-KJ37XZQA.js";
import {
  Alert,
  Spinner,
  useAsync
} from "./chunk-W6PEJALH.js";
import {
  require_react
} from "./chunk-NRAQAV6T.js";
import "./chunk-SQ3KPTIH.js";
import "./chunk-Z4MHKCET.js";
import "./chunk-C4SUTB4O.js";
import "./chunk-XWBYECTV.js";
import "./chunk-II7O6SCF.js";
import "./chunk-RL7RA2JG.js";
import "./chunk-PJCXC7XI.js";
import "./chunk-F43XHKVL.js";
import {
  require_lib
} from "./chunk-CU77QH5G.js";
import {
  __toESM,
  init_esm_shims
} from "./chunk-4EFJZ3GQ.js";

// ../cli-astro-commands/src/components/EnvRemoveCommand/index.ts
init_esm_shims();

// ../cli-astro-commands/src/components/EnvRemoveCommand/EnvRemoveCommand.tsx
init_esm_shims();
var import_react2 = __toESM(require_react(), 1);

// ../cli-astro-commands/src/components/EnvRemoveCommand/EnvRemove.tsx
init_esm_shims();
var import_react = __toESM(require_react(), 1);
var import_variant = __toESM(require_lib(), 1);
var ENV_NAME = "prod";
var EnvRemove = ({ variableKey }) => {
  const { model } = useProjectModel();
  const backendAsAServiceClient = useBackendAsAServiceClient();
  const { status } = useAsync(async () => {
    await backendAsAServiceClient.removeAppEnvironmentVariable(
      model.config.appId,
      ENV_NAME,
      variableKey
    );
  }, []);
  return (0, import_variant.match)(status, {
    Loading: () => /* @__PURE__ */ import_react.default.createElement(Spinner, { text: "Removing environment variable..." }),
    Success: () => /* @__PURE__ */ import_react.default.createElement(Alert, { type: "success" }, '"', variableKey, '" was successfully removed from "', ENV_NAME, `" environment. Note: This only removes the variable from the remote environment. It won't delete it from .env.local.`),
    Error: () => null,
    NotRequested: () => null
  });
};

// ../cli-astro-commands/src/components/EnvRemoveCommand/EnvRemoveCommand.tsx
var EnvRemoveCommand = ({ variableKey }) => {
  const {
    model: { config }
  } = useProjectModel();
  let commandTree = /* @__PURE__ */ import_react2.default.createElement(EnvRemove, { variableKey });
  if (config.projectType === "Site") {
    commandTree = /* @__PURE__ */ import_react2.default.createElement(SiteAuthProvider, { siteId: config.siteId }, commandTree);
  }
  return commandTree;
};
export {
  EnvRemoveCommand
};
