// @wix/cli/build/EnvSetCommand-M3COJ5U3.js
import { createRequire as _createRequire } from 'node:module';
const require = _createRequire(import.meta.url);
import {
  useBackendAsAServiceClient
} from "./chunk-TGRXPHY6.js";
import "./chunk-J5YVHTPQ.js";
import {
  useProjectModel
} from "./chunk-4ASXFAN7.js";
import "./chunk-KBORAL57.js";
import "./chunk-BVM3FYNM.js";
import {
  SiteAuthProvider
} from "./chunk-6PYC5DBT.js";
import "./chunk-WMCHVSMU.js";
import "./chunk-GUSV4N2E.js";
import "./chunk-KJ37XZQA.js";
import {
  Alert,
  Spinner,
  useAsync
} from "./chunk-LQXRNBJ2.js";
import {
  require_react
} from "./chunk-NRAQAV6T.js";
import "./chunk-SQ3KPTIH.js";
import "./chunk-Z4MHKCET.js";
import "./chunk-C4SUTB4O.js";
import "./chunk-P23NP3OP.js";
import "./chunk-II7O6SCF.js";
import "./chunk-RL7RA2JG.js";
import "./chunk-MDGDEDGX.js";
import "./chunk-F43XHKVL.js";
import {
  require_lib
} from "./chunk-STQCW7A5.js";
import {
  __toESM,
  init_esm_shims
} from "./chunk-4EFJZ3GQ.js";

// ../cli-astro-commands/src/components/EnvSetCommand/index.ts
init_esm_shims();

// ../cli-astro-commands/src/components/EnvSetCommand/EnvSetCommand.tsx
init_esm_shims();
var import_react2 = __toESM(require_react(), 1);

// ../cli-astro-commands/src/components/EnvSetCommand/EnvSet.tsx
init_esm_shims();
var import_react = __toESM(require_react(), 1);
var import_variant = __toESM(require_lib(), 1);
var ENV_NAME = "prod";
var EnvSet = ({
  variable
}) => {
  const { model } = useProjectModel();
  const backendAsAServiceClient = useBackendAsAServiceClient();
  const { status } = useAsync(async () => {
    await backendAsAServiceClient.upsertAppEnvironment(
      model.config.appId,
      ENV_NAME,
      {
        [variable.key]: variable.value
      }
    );
  }, []);
  return (0, import_variant.match)(status, {
    Loading: () => /* @__PURE__ */ import_react.default.createElement(Spinner, { text: "Setting environment variable..." }),
    Success: () => /* @__PURE__ */ import_react.default.createElement(Alert, { type: "success" }, '"', variable.key, '" was successfully set'),
    Error: () => null,
    NotRequested: () => null
  });
};

// ../cli-astro-commands/src/components/EnvSetCommand/EnvSetCommand.tsx
var EnvSetCommand = ({ variable }) => {
  const {
    model: {
      config: { siteId }
    }
  } = useProjectModel();
  return /* @__PURE__ */ import_react2.default.createElement(SiteAuthProvider, { siteId }, /* @__PURE__ */ import_react2.default.createElement(EnvSet, { variable }));
};
export {
  EnvSetCommand
};
