// @wix/cli/build/LogoutCommand-GSDL4UZG.js
import { createRequire as _createRequire } from 'node:module';
const require = _createRequire(import.meta.url);
import {
  logout
} from "./chunk-6PYC5DBT.js";
import "./chunk-WMCHVSMU.js";
import "./chunk-GUSV4N2E.js";
import "./chunk-KJ37XZQA.js";
import {
  Box_default,
  ConfirmInput,
  Spinner,
  Text,
  useAsyncCallback
} from "./chunk-LQXRNBJ2.js";
import {
  require_react
} from "./chunk-NRAQAV6T.js";
import "./chunk-SQ3KPTIH.js";
import "./chunk-Z4MHKCET.js";
import "./chunk-C4SUTB4O.js";
import "./chunk-II7O6SCF.js";
import "./chunk-RL7RA2JG.js";
import "./chunk-MDGDEDGX.js";
import "./chunk-F43XHKVL.js";
import {
  require_lib
} from "./chunk-STQCW7A5.js";
import {
  __toESM,
  init_esm_shims
} from "./chunk-4EFJZ3GQ.js";

// src/components/LogoutCommand/index.ts
init_esm_shims();

// src/components/LogoutCommand/LogoutCommand.tsx
init_esm_shims();
var import_react = __toESM(require_react(), 1);
var import_variant = __toESM(require_lib(), 1);
var LogoutCommand = ({ userInfo }) => {
  const logoutAction = useAsyncCallback(logout);
  if (userInfo == null) {
    return /* @__PURE__ */ import_react.default.createElement(Text, null, "Wix CLI is already logged out");
  }
  return /* @__PURE__ */ import_react.default.createElement(import_react.default.Fragment, null, /* @__PURE__ */ import_react.default.createElement(Box_default, { marginBottom: 1 }, /* @__PURE__ */ import_react.default.createElement(Text, null, "When you're logged out you cannot use Wix CLI commands")), /* @__PURE__ */ import_react.default.createElement(
    ConfirmInput,
    {
      label: "Proceed with logout?",
      initialValue: true,
      onSubmit: (confirmed) => {
        if (confirmed) {
          void logoutAction.execute();
        }
      }
    }
  ), (0, import_variant.match)(logoutAction.status, {
    NotRequested: () => null,
    Error: () => null,
    Loading: () => /* @__PURE__ */ import_react.default.createElement(Spinner, null),
    Success: () => /* @__PURE__ */ import_react.default.createElement(Text, null, "Wix CLI is logged out")
  }));
};
export {
  LogoutCommand
};
