// @wix/cli/build/PreviewCommand-7NPU4ZHT.js
import { createRequire as _createRequire } from 'node:module';
const require = _createRequire(import.meta.url);
import {
  useCreatePreview
} from "./chunk-HZCGDSWJ.js";
import {
  DeploymentStatus
} from "./chunk-XIMJA744.js";
import "./chunk-6CKGZQ6Y.js";
import "./chunk-CH52GF36.js";
import "./chunk-DBD3AN3F.js";
import "./chunk-VJSDBG2C.js";
import "./chunk-YMAQMJWZ.js";
import "./chunk-74NPPELF.js";
import "./chunk-LG2AMEKY.js";
import "./chunk-MVBGFP6D.js";
import "./chunk-PSBMQW7U.js";
import "./chunk-4ANLJVRU.js";
import "./chunk-C6TNYPLG.js";
import "./chunk-BKY3IKB3.js";
import {
  ErrorViewer
} from "./chunk-SZ4GCQYL.js";
import "./chunk-WMCHVSMU.js";
import {
  Trans
} from "./chunk-6C4RLYLU.js";
import "./chunk-KJ37XZQA.js";
import {
  Alert,
  Badge,
  Box_default,
  Link,
  Spinner,
  useAsync
} from "./chunk-W6PEJALH.js";
import {
  require_react
} from "./chunk-NRAQAV6T.js";
import "./chunk-SQ3KPTIH.js";
import "./chunk-Z4MHKCET.js";
import "./chunk-C4SUTB4O.js";
import "./chunk-XWBYECTV.js";
import "./chunk-II7O6SCF.js";
import "./chunk-RL7RA2JG.js";
import "./chunk-PJCXC7XI.js";
import "./chunk-F43XHKVL.js";
import {
  require_lib
} from "./chunk-CU77QH5G.js";
import {
  __toESM,
  init_esm_shims
} from "./chunk-4EFJZ3GQ.js";

// ../cli-site/src/components/PreviewCommand/PreviewCommand.tsx
init_esm_shims();
var import_react = __toESM(require_react(), 1);
var import_variant = __toESM(require_lib(), 1);
var PreviewCommand = ({ force }) => {
  const createPreview = useCreatePreview();
  const { status } = useAsync(async () => {
    const { previewUrl, deploymentStatus } = await createPreview();
    if (!force && (0, import_variant.isType)(deploymentStatus, DeploymentStatus.Error)) {
      throw deploymentStatus.deploymentError;
    }
    return { previewUrl, deploymentStatus };
  }, []);
  return (0, import_variant.match)(status, {
    Error: () => null,
    Loading: () => /* @__PURE__ */ import_react.default.createElement(Spinner, { text: /* @__PURE__ */ import_react.default.createElement(Trans, { i18nKey: "preview_command.loading" }) }),
    Success: ({ result: { previewUrl, deploymentStatus } }) => /* @__PURE__ */ import_react.default.createElement(import_react.default.Fragment, null, (0, import_variant.isType)(deploymentStatus, DeploymentStatus.Error) ? /* @__PURE__ */ import_react.default.createElement(Box_default, { marginBottom: 1, flexDirection: "column" }, /* @__PURE__ */ import_react.default.createElement(ErrorViewer, { error: deploymentStatus.deploymentError })) : null, /* @__PURE__ */ import_react.default.createElement(Alert, { type: "success" }, /* @__PURE__ */ import_react.default.createElement(Badge, { skin: "success" }, /* @__PURE__ */ import_react.default.createElement(Trans, { i18nKey: "general.success" })), " ", /* @__PURE__ */ import_react.default.createElement(
      Trans,
      {
        i18nKey: "preview_command.success",
        components: [/* @__PURE__ */ import_react.default.createElement(Link, { url: previewUrl })]
      }
    )))
  });
};
export {
  PreviewCommand
};
