// @wix/cli/build/PreviewCommand-63Y2QMZ3.js
import { createRequire as _createRequire } from 'node:module';
const require = _createRequire(import.meta.url);
import {
  useCreatePreview
} from "./chunk-USA5LODZ.js";
import {
  DeploymentStatus
} from "./chunk-UR2HUVJ7.js";
import "./chunk-GJRDQFCB.js";
import "./chunk-CH52GF36.js";
import "./chunk-E7PJXDFO.js";
import "./chunk-GGTDDK53.js";
import "./chunk-V6HQQWCA.js";
import "./chunk-LUM7NYOU.js";
import "./chunk-U7NMQMEG.js";
import "./chunk-MVBGFP6D.js";
import "./chunk-MDMFU25C.js";
import "./chunk-3MBMJERV.js";
import "./chunk-KBORAL57.js";
import "./chunk-BVM3FYNM.js";
import {
  ErrorViewer
} from "./chunk-6PYC5DBT.js";
import "./chunk-WMCHVSMU.js";
import {
  Trans
} from "./chunk-GUSV4N2E.js";
import "./chunk-KJ37XZQA.js";
import {
  Alert,
  Badge,
  Box_default,
  Link,
  Spinner,
  useAsync
} from "./chunk-LQXRNBJ2.js";
import {
  require_react
} from "./chunk-NRAQAV6T.js";
import "./chunk-SQ3KPTIH.js";
import "./chunk-Z4MHKCET.js";
import "./chunk-C4SUTB4O.js";
import "./chunk-P23NP3OP.js";
import "./chunk-II7O6SCF.js";
import "./chunk-RL7RA2JG.js";
import "./chunk-MDGDEDGX.js";
import "./chunk-F43XHKVL.js";
import {
  require_lib
} from "./chunk-STQCW7A5.js";
import {
  __toESM,
  init_esm_shims
} from "./chunk-4EFJZ3GQ.js";

// ../cli-site/src/components/PreviewCommand/PreviewCommand.tsx
init_esm_shims();
var import_react = __toESM(require_react(), 1);
var import_variant = __toESM(require_lib(), 1);
var PreviewCommand = ({ force }) => {
  const createPreview = useCreatePreview();
  const { status } = useAsync(async () => {
    const { previewUrl, deploymentStatus } = await createPreview();
    if (!force && (0, import_variant.isType)(deploymentStatus, DeploymentStatus.Error)) {
      throw deploymentStatus.deploymentError;
    }
    return { previewUrl, deploymentStatus };
  }, []);
  return (0, import_variant.match)(status, {
    Error: () => null,
    Loading: () => /* @__PURE__ */ import_react.default.createElement(Spinner, { text: /* @__PURE__ */ import_react.default.createElement(Trans, { i18nKey: "preview_command.loading" }) }),
    Success: ({ result: { previewUrl, deploymentStatus } }) => /* @__PURE__ */ import_react.default.createElement(import_react.default.Fragment, null, (0, import_variant.isType)(deploymentStatus, DeploymentStatus.Error) ? /* @__PURE__ */ import_react.default.createElement(Box_default, { marginBottom: 1, flexDirection: "column" }, /* @__PURE__ */ import_react.default.createElement(ErrorViewer, { error: deploymentStatus.deploymentError })) : null, /* @__PURE__ */ import_react.default.createElement(Alert, { type: "success" }, /* @__PURE__ */ import_react.default.createElement(Badge, { skin: "success" }, /* @__PURE__ */ import_react.default.createElement(Trans, { i18nKey: "general.success" })), " ", /* @__PURE__ */ import_react.default.createElement(
      Trans,
      {
        i18nKey: "preview_command.success",
        components: [/* @__PURE__ */ import_react.default.createElement(Link, { url: previewUrl })]
      }
    )))
  });
};
export {
  PreviewCommand
};
