// @wix/cli/build/PromoteCommand-SU2X566V.js
import { createRequire as _createRequire } from 'node:module';
const require = _createRequire(import.meta.url);
import {
  PreviewCommand
} from "./chunk-6B2RYJDL.js";
import {
  useARMClient
} from "./chunk-6QHN2XDV.js";
import "./chunk-CUQUD7FS.js";
import "./chunk-RWTQFIPN.js";
import "./chunk-W54PG6O5.js";
import "./chunk-TGRXPHY6.js";
import "./chunk-J5YVHTPQ.js";
import {
  useProjectModel
} from "./chunk-4ASXFAN7.js";
import "./chunk-KBORAL57.js";
import "./chunk-PIXNEJ52.js";
import "./chunk-HWVLWVTF.js";
import "./chunk-BVM3FYNM.js";
import "./chunk-6PYC5DBT.js";
import "./chunk-WMCHVSMU.js";
import "./chunk-GUSV4N2E.js";
import "./chunk-KJ37XZQA.js";
import {
  Alert,
  Box_default,
  Spinner,
  Text,
  useAsyncCallback
} from "./chunk-LQXRNBJ2.js";
import {
  require_react
} from "./chunk-NRAQAV6T.js";
import "./chunk-SQ3KPTIH.js";
import "./chunk-Z4MHKCET.js";
import "./chunk-C4SUTB4O.js";
import "./chunk-P23NP3OP.js";
import "./chunk-II7O6SCF.js";
import "./chunk-RL7RA2JG.js";
import "./chunk-MDGDEDGX.js";
import "./chunk-F43XHKVL.js";
import {
  require_lib
} from "./chunk-STQCW7A5.js";
import {
  __toESM,
  init_esm_shims
} from "./chunk-4EFJZ3GQ.js";

// ../cli-astro-commands/src/components/PromoteCommand/index.ts
init_esm_shims();

// ../cli-astro-commands/src/components/PromoteCommand/PromoteCommand.tsx
init_esm_shims();
var import_react2 = __toESM(require_react(), 1);
var import_variant = __toESM(require_lib(), 1);

// ../cli-astro-commands/src/components/PromoteCommand/Steps/CreatePreview.tsx
init_esm_shims();
var import_react = __toESM(require_react(), 1);
var CreatePreview = (props) => {
  const { baseUrl, label, onCreated } = props;
  return /* @__PURE__ */ import_react.default.createElement(Box_default, { flexDirection: "column", gap: 1 }, /* @__PURE__ */ import_react.default.createElement(Text, null, "Before promoting, let's create a preview of your app"), /* @__PURE__ */ import_react.default.createElement(Box_default, { flexDirection: "column" }, /* @__PURE__ */ import_react.default.createElement(
    PreviewCommand,
    {
      baseUrl,
      label,
      onPreviewCreated: onCreated,
      environment: "preview"
    }
  )));
};

// ../cli-astro-commands/src/components/PromoteCommand/PromoteCommand.tsx
var PromoteCommand = ({ baseUrl, label }) => {
  const {
    model: {
      config: { appId }
    }
  } = useProjectModel();
  const arm = useARMClient();
  const promoteAction = useAsyncCallback(async (_, tag) => {
    await arm.createDraftVersion(appId, tag);
  });
  function onPreviewCreated({ previewId }) {
    return promoteAction.execute(previewId);
  }
  return /* @__PURE__ */ import_react2.default.createElement(import_react2.default.Fragment, null, /* @__PURE__ */ import_react2.default.createElement(
    CreatePreview,
    {
      baseUrl,
      label,
      onCreated: ({ previewId }) => onPreviewCreated({ previewId })
    }
  ), (0, import_variant.match)(promoteAction.status, {
    Loading: () => /* @__PURE__ */ import_react2.default.createElement(Spinner, { text: `Promoting app draft...` }),
    Success: () => /* @__PURE__ */ import_react2.default.createElement(Alert, { type: "success" }, "App draft promoted successfully"),
    Error: () => null,
    NotRequested: () => null
  }));
};
export {
  PromoteCommand
};
