// @wix/cli/build/uninstall-UKE4WSXR.js
import { createRequire as _createRequire } from 'node:module';
const require = _createRequire(import.meta.url);
import {
  resolveNpmDependencies,
  resolvePackageManager
} from "./chunk-V3ZO6GY5.js";
import "./chunk-5AY2TF4R.js";
import {
  createLogger
} from "./chunk-X2NBNP2E.js";
import "./chunk-XNSO7WKJ.js";
import {
  createI18nT
} from "./chunk-PNV2OKNW.js";
import "./chunk-HEANTKSX.js";
import "./chunk-6EJNAOBP.js";
import {
  authenticate
} from "./chunk-T5C43OQK.js";
import "./chunk-CH52GF36.js";
import "./chunk-MDMFU25C.js";
import "./chunk-MB2DJ4WP.js";
import "./chunk-CUQUD7FS.js";
import "./chunk-RWTQFIPN.js";
import "./chunk-W54PG6O5.js";
import "./chunk-ZBDV4YE5.js";
import "./chunk-6PYC5DBT.js";
import "./chunk-WMCHVSMU.js";
import "./chunk-GUSV4N2E.js";
import "./chunk-KJ37XZQA.js";
import "./chunk-LQXRNBJ2.js";
import "./chunk-NRAQAV6T.js";
import "./chunk-SQ3KPTIH.js";
import "./chunk-Z4MHKCET.js";
import "./chunk-C4SUTB4O.js";
import "./chunk-P23NP3OP.js";
import "./chunk-II7O6SCF.js";
import "./chunk-RL7RA2JG.js";
import "./chunk-MDGDEDGX.js";
import "./chunk-F43XHKVL.js";
import {
  CliError,
  CliErrorCode
} from "./chunk-STQCW7A5.js";
import {
  init_esm_shims
} from "./chunk-4EFJZ3GQ.js";

// ../cli-site-old/src/uninstall/index.ts
init_esm_shims();

// ../cli-site-old/src/uninstall/uninstall.ts
init_esm_shims();
async function uninstall(model, services, options, initialPackageName) {
  const t = createI18nT();
  const logger = createLogger({ t });
  const authState = await authenticate({
    metaSiteId: model.metaSiteId,
    biLogger: services.biLogger,
    errorReporter: services.errorReporter
  });
  const packageName = initialPackageName.trim();
  if (!packageName) {
    throw new CliError({
      code: CliErrorCode.WixUninstallMissingPackageName(),
      cause: null
    });
  }
  const logDependencyUninstalled = logger.logDependencyUninstalling(packageName);
  try {
    const packageManager = await resolvePackageManager(
      model.projectFolder,
      options
    );
    await packageManager.runUninstallPackage(model.projectFolder, packageName);
    await resolveNpmDependencies(model.projectFolder, { authState });
    logDependencyUninstalled.success();
  } catch (error) {
    logDependencyUninstalled.fail();
    throw error;
  }
}
export {
  uninstall
};
