// @wix/cli/build/uninstall-Q5N7ZWG7.js
import { createRequire as _createRequire } from 'node:module';
const require = _createRequire(import.meta.url);
import {
  resolveNpmDependencies,
  resolvePackageManager
} from "./chunk-YCSOXW4E.js";
import "./chunk-5AY2TF4R.js";
import {
  createLogger
} from "./chunk-6RNNDF2T.js";
import "./chunk-BQDKSF4L.js";
import {
  createI18nT
} from "./chunk-DZWD6YGQ.js";
import "./chunk-HEANTKSX.js";
import "./chunk-6EJNAOBP.js";
import {
  authenticate
} from "./chunk-A6BVUV6V.js";
import "./chunk-CH52GF36.js";
import "./chunk-PSBMQW7U.js";
import "./chunk-IRKCE3VI.js";
import "./chunk-ZWAKRWC6.js";
import "./chunk-2R5BII2J.js";
import "./chunk-W54PG6O5.js";
import "./chunk-YLWTNITT.js";
import "./chunk-EVROMEWT.js";
import "./chunk-SZ4GCQYL.js";
import "./chunk-WMCHVSMU.js";
import "./chunk-6C4RLYLU.js";
import "./chunk-KJ37XZQA.js";
import "./chunk-W6PEJALH.js";
import "./chunk-NRAQAV6T.js";
import "./chunk-SQ3KPTIH.js";
import "./chunk-Z4MHKCET.js";
import "./chunk-C4SUTB4O.js";
import "./chunk-XWBYECTV.js";
import "./chunk-II7O6SCF.js";
import "./chunk-RL7RA2JG.js";
import "./chunk-PJCXC7XI.js";
import "./chunk-F43XHKVL.js";
import {
  CliError,
  CliErrorCode
} from "./chunk-CU77QH5G.js";
import {
  init_esm_shims
} from "./chunk-4EFJZ3GQ.js";

// ../cli-site-old/src/uninstall/index.ts
init_esm_shims();

// ../cli-site-old/src/uninstall/uninstall.ts
init_esm_shims();
async function uninstall(model, services, options, initialPackageName) {
  const t = createI18nT();
  const logger = createLogger({ t });
  const authState = await authenticate({
    metaSiteId: model.metaSiteId,
    biLogger: services.biLogger,
    errorReporter: services.errorReporter
  });
  const packageName = initialPackageName.trim();
  if (!packageName) {
    throw new CliError({
      code: CliErrorCode.WixUninstallMissingPackageName(),
      cause: null
    });
  }
  const logDependencyUninstalled = logger.logDependencyUninstalling(packageName);
  try {
    const packageManager = await resolvePackageManager(
      model.projectFolder,
      options
    );
    await packageManager.runUninstallPackage(model.projectFolder, packageName);
    await resolveNpmDependencies(model.projectFolder, { authState });
    logDependencyUninstalled.success();
  } catch (error) {
    logDependencyUninstalled.fail();
    throw error;
  }
}
export {
  uninstall
};
