// @wix/cli/build/UninstallCommand-ZASEMR45.js
import { createRequire as _createRequire } from 'node:module';
const require = _createRequire(import.meta.url);
import {
  createDependenciesService
} from "./chunk-LAVKWSGY.js";
import "./chunk-GGTDDK53.js";
import {
  useVeloClient
} from "./chunk-V6HQQWCA.js";
import "./chunk-LUM7NYOU.js";
import "./chunk-U7NMQMEG.js";
import "./chunk-MVBGFP6D.js";
import "./chunk-MDMFU25C.js";
import {
  useProjectModel
} from "./chunk-3MBMJERV.js";
import "./chunk-CUQUD7FS.js";
import "./chunk-RWTQFIPN.js";
import "./chunk-W54PG6O5.js";
import "./chunk-KBORAL57.js";
import "./chunk-BVM3FYNM.js";
import "./chunk-6PYC5DBT.js";
import "./chunk-WMCHVSMU.js";
import {
  Trans
} from "./chunk-GUSV4N2E.js";
import "./chunk-KJ37XZQA.js";
import {
  Alert,
  Badge,
  Spinner,
  Text,
  useAsync
} from "./chunk-LQXRNBJ2.js";
import {
  require_react
} from "./chunk-NRAQAV6T.js";
import "./chunk-SQ3KPTIH.js";
import "./chunk-Z4MHKCET.js";
import "./chunk-C4SUTB4O.js";
import "./chunk-P23NP3OP.js";
import "./chunk-II7O6SCF.js";
import "./chunk-RL7RA2JG.js";
import "./chunk-MDGDEDGX.js";
import "./chunk-F43XHKVL.js";
import {
  require_lib
} from "./chunk-STQCW7A5.js";
import {
  __toESM,
  init_esm_shims
} from "./chunk-4EFJZ3GQ.js";

// ../cli-site/src/components/UninstallCommand/UninstallCommand.tsx
init_esm_shims();
var import_react = __toESM(require_react(), 1);
var import_variant = __toESM(require_lib(), 1);
var UninstallCommand = ({ packageName }) => {
  const { model } = useProjectModel();
  const veloClient = useVeloClient();
  const { status } = useAsync(async () => {
    const { projectFolder } = model;
    const service = await createDependenciesService(projectFolder, veloClient);
    await service.uninstall(packageName);
  }, []);
  return (0, import_variant.match)(status, {
    Loading: () => /* @__PURE__ */ import_react.default.createElement(
      Spinner,
      {
        text: /* @__PURE__ */ import_react.default.createElement(
          Trans,
          {
            i18nKey: "uninstall_command.loading",
            values: { packageName },
            components: [/* @__PURE__ */ import_react.default.createElement(Text, { bold: true })]
          }
        )
      }
    ),
    Error: () => null,
    Success: () => /* @__PURE__ */ import_react.default.createElement(Alert, { type: "success" }, /* @__PURE__ */ import_react.default.createElement(Badge, { skin: "success" }, /* @__PURE__ */ import_react.default.createElement(Trans, { i18nKey: "general.success" })), " ", /* @__PURE__ */ import_react.default.createElement(
      Trans,
      {
        i18nKey: "uninstall_command.success",
        values: { packageName },
        components: [/* @__PURE__ */ import_react.default.createElement(Text, { bold: true })]
      }
    ))
  });
};
export {
  UninstallCommand
};
